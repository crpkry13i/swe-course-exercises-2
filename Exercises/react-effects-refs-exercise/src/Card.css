.Card {
  position: absolute;
  z-index: 100;
  left: 0;
  right: 0;
  margin: auto;
}
