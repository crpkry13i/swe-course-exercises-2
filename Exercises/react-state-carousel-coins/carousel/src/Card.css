.Card {
  display: flex;
  flex-direction: column;
}

.Card-title {
  margin: 0 0 0.5em 0;
}

.Card-small {
  margin-top: 0.5em;
  font-style: italic;
}

.Card-image {
  border-radius: 8px;
}