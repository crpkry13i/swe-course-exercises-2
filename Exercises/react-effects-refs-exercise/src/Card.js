import React, { useState } from "react";
import './Card.css';

/** Single card: just renders the card as received from deck. */

function Card({name, image}) {

  const [{angle, xPos, yPos}] = useState({
    angle: Math.random() * 90 - 45,
    xPos: Math.random() * 40 - 20,
    yPos: Math.random() * 40 - 20
  });

  const transform = `translate(${xPos}px, ${yPos}px) rotate(${angle}deg)`;

  return <img className="Card"
            alt={name}
            src={image}
            style={{transform}} />;
}

export default Card
