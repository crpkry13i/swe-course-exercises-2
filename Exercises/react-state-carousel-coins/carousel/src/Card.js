import React from "react";
import "./Card.css";

function Card(props) {
  return (
    <div className="Card">
      <h4 className="Card-title">{props.caption}</h4>
      <img className="Card-image" src={props.src} alt={props.caption} />
      <small className="Card-small">
        Image {props.currNum} of {props.totalNum}.
      </small>
    </div>
  );
}

export default Card;
