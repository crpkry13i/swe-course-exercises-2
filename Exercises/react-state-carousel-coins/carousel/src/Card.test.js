import React from "react";
import { render } from "@testing-library/react";
import Card from "./Card";

// smoke test
it("renders without crashing", () => {
  render(<Card />);
});

// snapshot test
it("matches snapshot", () => {
  const { asFragment } = render(<Card />);
  expect(asFragment()).toMatchSnapshot();
});
