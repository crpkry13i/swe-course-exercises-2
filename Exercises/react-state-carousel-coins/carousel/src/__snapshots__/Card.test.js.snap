// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    />
    <img
      class="Card-image"
    />
    <small
      class="Card-small"
    >
      Image  of .
    </small>
  </div>
</DocumentFragment>
`;
